src/commands/install.rs

```diff
@@ -1,113 +1,215 @@
-use std::fmt::Write;
-use std::fs;
-
 use anyhow::{bail, Result};
 use clap::Parser;
+use regex::Regex;
+use std::fmt::Write;
+use std::fs;
 use tracing::{error, info};
 
-use crate::constants::{FUEL_CORE_RELEASE_DOWNLOAD_URL, SWAY_RELEASE_DOWNLOAD_URL};
-use crate::download::{download_file_and_unpack, fuelup_bin_dir, unpack_extracted_bins};
-use crate::{
-    constants::{FUEL_CORE_REPO, GITHUB_API_REPOS_BASE_URL, RELEASES_LATEST, SWAY_REPO},
-    download::{forc_bin_tarball_name, fuel_core_bin_tarball_name, get_latest_tag},
+use crate::download::{
+    download_file_and_unpack, fuelup_bin_dir, unpack_extracted_bins, DownloadCfg,
 };
 
 #[derive(Debug, Parser)]
-pub struct InstallCommand {}
+#[clap(override_usage = "\
+    fuelup install <COMPONENT>[@<VERSION>] ...")]
+pub struct InstallCommand {
+    /// Reference to a forc component to add as a dependency
+    ///
+    /// You can reference components by:{n}
+    /// - `<name>`, like `fuelup install forc` (latest version will be used){n}
+    /// - `<name>@<version>`, like `fuelup install forc@0.14.5`{n}
+    #[clap(multiple_values = true)]
+    components: Vec<String>,
+}
 
-pub fn install() -> Result<()> {
-    info!("\nDownloading the Fuel toolchain\n");
+pub const POSSIBLE_COMPONENTS: [&str; 3] = ["forc", "fuel-core", "fuelup"];
 
-    let forc_release_latest_tag = match get_latest_tag(&format!(
-        "{}{}/{}",
-        GITHUB_API_REPOS_BASE_URL, SWAY_REPO, RELEASES_LATEST
-    )) {
-        Ok(t) => t,
-        Err(_) => bail!("Failed to fetch latest forc release tag from GitHub API"),
+pub fn install_one(download_cfg: DownloadCfg) -> Result<DownloadCfg> {
+    let fuelup_bin_dir = fuelup_bin_dir();
+    if !fuelup_bin_dir.is_dir() {
+        fs::create_dir_all(&fuelup_bin_dir).expect("Unable to create fuelup directory");
+    }
+
+    info!("Fetching {} {}", &download_cfg.name, &download_cfg.version);
+
+    if download_file_and_unpack(&download_cfg).is_err() {
+        bail!("{} {}", &download_cfg.name, &download_cfg.version)
     };
 
-    let fuel_core_release_latest_tag = match get_latest_tag(&format!(
-        "{}{}/{}",
-        GITHUB_API_REPOS_BASE_URL, FUEL_CORE_REPO, RELEASES_LATEST
-    )) {
-        Ok(t) => t,
-        Err(_) => bail!("Failed to fetch latest fuel-core release tag from GitHub API"),
+    if unpack_extracted_bins(&fuelup_bin_dir).is_err() {
+        bail!("{} {}", &download_cfg.name, &download_cfg.version)
     };
 
-    let fuelup_bin_dir = fuelup_bin_dir();
-    if !fuelup_bin_dir.is_dir() {
-        fs::create_dir_all(&fuelup_bin_dir)?;
+    Ok(download_cfg)
+}
+
+pub fn parse_component(component: &str) -> Result<(String, Option<String>)> {
+    if component.contains('@') {
+        let filtered = component.split('@').collect::<Vec<&str>>();
+        let semver_regex = Regex::new(r"^[v]?\d+\.\d+\.\d+$").unwrap();
```

Comment:
Perhaps rather than introducing `regex`, we could use the `semver` crate? E.g. `semver::Version::parse(version)`? I'm thinking it might be a little more light-weight / easier to read for non-regex magicians.

Reply:
you're right, this would definitely be alot more legible and would help with #64 as well (since we're checking versions), seems like it also handles validation which is everything we need for this